### PR TITLE
Fix NPE in registeredLookup extractionFn when "optimize" is not provided.

### PR DESCRIPTION
**Pull request by gianm**

### CODE REVIEW

File: processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public RegisteredLookupExtractionFn(
     this.replaceMissingValueWith = replaceMissingValueWith;
     this.retainMissingValue = retainMissingValue;
     this.injective = injective;
-    this.optimize = optimize;
+    this.optimize = optimize == null ? true : optimize;
```
Comment (drcrallen) [inline processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java]: optimize is an experimental path, the input variable should be `boolean`

Comment (gianm) [inline processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java]: This is just mimicking the behavior in LookupDimensionSpec.

Comment (gianm) [inline processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java]: I take no stand on what the default should be, other than that they should behave the same way. Happy to change them both to default false.

**Comment by gianm** [comment]
travis fail due to #3067

File: processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java (hunk 1 above)
Comment (drcrallen) [inline processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java]: arg... consistency for now I agree is better.

**Comment by drcrallen** [comment]
👍

**Comment by nishantmonu51** [comment]
:+1:, after travis

**Comment by gianm** [comment]
https://travis-ci.org/druid-io/druid/jobs/134941003

```
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 60.125 sec <<< FAILURE! - in io.druid.server.QueryResourceTest
testSecuredGetServer(io.druid.server.QueryResourceTest)  Time elapsed: 60.022 sec  <<< ERROR!
java.lang.Exception: test timed out after 60000 milliseconds
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
	at io.druid.server.QueryResourceTest.testSecuredGetServer(QueryResourceTest.java:323)
```

**Comment by gianm** [comment]
https://travis-ci.org/druid-io/druid/builds/135057242 (#2253)

```
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 66.962 sec <<< FAILURE! - in io.druid.curator.announcement.AnnouncerTest
testSanity(io.druid.curator.announcement.AnnouncerTest)  Time elapsed: 61.106 sec  <<< ERROR!
java.lang.Exception: test timed out after 60000 milliseconds
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:226)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedNanos(AbstractQueuedSynchronizer.java:1033)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.tryAcquireSharedNanos(AbstractQueuedSynchronizer.java:1326)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:282)
	at org.apache.curator.test.Timing.awaitLatch(Timing.java:120)
	at io.druid.curator.announcement.AnnouncerTest.testSanity(AnnouncerTest.java:101)
```

**Pull request merged**